Source/ASTextNode.mm

```diff
@@ -1047,7 +1048,8 @@ - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event
   NSUInteger lastCharIndex = NSIntegerMax;
   BOOL linkCrossesVisibleRange = (lastCharIndex > range.location) && (lastCharIndex < NSMaxRange(range) - 1);
 
-  if (range.length > 0 && !linkCrossesVisibleRange && linkAttributeValue != nil && linkAttributeName != nil) {
+  if ((range.length > 0 && !linkCrossesVisibleRange && linkAttributeValue != nil && linkAttributeName != nil) ||
+    _alwaysHandleTruncationTokenTap) {
```

Comment:
Nit: if we always return `YES` when `_alwaysHandleTruncationTokenTap` is set then I think it's better to return as early as possible. So we can do this [here](https://github.com/TextureGroup/Texture/pull/1520/files#diff-271d6b828178013b300f794e8ec8c740R1037):

```
if (_alwaysHandleTruncationTokenTap) {
  return YES;
}
```

Reply:
Another thing: we don't acquire the lock while accessing/setting `_alwaysHandleTruncationTokenTap`. We should fix that.

Same deal for `_passthroughNonlinkTouches` and `delegate`. It'd be great if you can help to fix them in a follow-up PR.

Reply:
@nguyenhuy done, I will add the lock of `_passthroughNonlinkTouches` and `delegate` in a follow-up PR.